src/file_part.rs

```diff
@@ -0,0 +1,180 @@
+use crate::errors::AppError;
+use crate::json_part::read_page_to_json_str_headlines;
+use crate::message_part::send_first_upd;
+use log::{error, info};
+use serde_json::{self, Value};
+use std::fs;
+use std::fs::File;
+use std::io::Read;
+use std::io::Write;
+
+const FILE1: &str = "temp_new.json";
+const FILE2: &str = "temp_old.json";
+
+/// Writes headlines to a JSON file.
+///
+/// This function writes the provided headlines to a JSON file. It converts the headlines into
+/// a JSON string using `serde_json::to_string()` and writes the string to the specified file.
+/// Returns `Ok(())` if the operation succeeds, otherwise returns an `io::Error`.
+pub async fn write_headlines_to_json_file(headlines: Vec<String>) -> Result<(), AppError> {
+    info!("Writing headlines to JSON file.");
+    let json_str = serde_json::to_string(&headlines).map_err(AppError::ParseJsonError)?;
+
+    let mut file = File::create("temp_new.json")?;
+    file.write_all(json_str.as_bytes())?;
+    info!("Headlines successfully written to JSON file.");
+
+    Ok(())
+}
+
+/// Reads the content of a file into a string.
+///
+/// This function reads the content of the specified file into a string. Returns `Ok(content)`
+/// if the operation succeeds, otherwise returns an `io::Error`.
+fn read_file_content(filename: &str) -> Result<String, AppError> {
+    info!("Reading content from file: {}", filename);
+    let mut file = File::open(filename).map_err(AppError::IoError)?;
+    let mut content = String::new();
+    file.read_to_string(&mut content)
+        .map_err(AppError::IoError)?;
+    info!("Content read from file: {}", filename);
+    Ok(content)
+}
+
+/// Parses a JSON string into a `serde_json::Value`.
+///
+/// This function parses the provided JSON string into a `serde_json::Value`. Returns `Ok(value)`
+/// if the operation succeeds, otherwise returns a `serde_json::Error`.
+fn parse_json(content: &str) -> Result<Value, serde_json::Error> {
+    info!("Parsing JSON.");
+    serde_json::from_str(content)
+}
+
+/// Compares two JSON values recursively.
+///
+/// This function compares two JSON values recursively. It returns `true` if the values are equal,
+/// otherwise returns `false`.
+fn compare_json_values(value1: &Value, value2: &Value) -> bool {
```

Comment:
How is this different from built-in derive for `Eq` for `serde_json::Value`?
https://docs.rs/serde_json/latest/serde_json/enum.Value.html#impl-Eq-for-Value